### PR TITLE
fix(test): replace flaky 1ms sleep with polling in engine test

### PR DESCRIPTION
**Pull request by geyslan**
### 1. Explain what the PR does

022cbe84a **fix(test): replace flaky 1ms sleep with polling in engine test**

> The TestEngine_ConsumeSources test relied on a 1ms sleep for event
> propagation, causing intermittent failures on loaded x86_64 runners
> when the engine goroutine hadn't processed the event in time.
> 
> Use require.Eventually (5s timeout, 10ms poll) with exact count
> matching for positive-event cases, and require.Never over a 50ms
> window for zero-event assertions. Remove dead gotEvent/emptyEvent
> comparison that always passed regardless of behavior.

--

### CODE REVIEW

File: pkg/signatures/engine/engine_test.go [hunk 1]
```diff
@@ -369,16 +362,14 @@ func TestEngine_ConsumeSources(t *testing.T) {
...
-				assert.Equal(t, emptyEvent, gotEvent, tc.name)
-				assert.Zero(t, eventsCount, tc.name)
+				require.Never(t, func() bool {
+					return atomic.LoadUint32(&gotNumEvents) > 0
+				}, 50*time.Millisecond, 10*time.Millisecond, tc.name)
 			} else {
-				assert.Equal(t, tc.expectedNumEvents, eventsCount, tc.name)
+				require.Eventually(t, func() bool {
+					return int(atomic.LoadUint32(&gotNumEvents)) == tc.expectedNumEvents
+				}, 5*time.Second, 10*time.Millisecond, tc.name)
```
[the remark this answers is not in the document]
Comment (geyslan) [reply pkg/signatures/engine/engine_test.go:372]: The == condition already handles the overshoot case deterministically. If the engine delivers 2 events before our first 10ms poll, the counter goes from 0 -> 1 -> 2, and since count == 1 is never true again (atomic counter is monotonically increasing), the test times out after 5s and fails.

The only undetected scenario is a duplicate that arrives in the nanoseconds between require.Eventually observing count == 1 and the function returning. Adding a 50ms require.Never post-condition to every positive test case (adding ~50ms per subtest) for an essentially impossible race in a unit test is not justified.

File: pkg/signatures/engine/engine_test.go [hunk 2]
```diff
@@ -369,16 +362,14 @@ func TestEngine_ConsumeSources(t *testing.T) {
...
-			var gotEvent protocol.Event
-			time.Sleep(time.Millisecond * 1) // wait for events to propagate
-
-			eventsCount := int(atomic.LoadUint32(&gotNumEvents))
 			if tc.expectedNumEvents <= 0 {
-				assert.Equal(t, emptyEvent, gotEvent, tc.name)
-				assert.Zero(t, eventsCount, tc.name)
+				require.Never(t, func() bool {
+					return atomic.LoadUint32(&gotNumEvents) > 0
+				}, 50*time.Millisecond, 10*time.Millisecond, tc.name)
```
[the remark this answers is not in the document]
Comment (geyslan) [reply pkg/signatures/engine/engine_test.go:368]: The event is written to a buffered channel (size 1), and the engine goroutine is already running a tight for event := range inputs.Tracee loop. The channel read + selector mismatch check takes microseconds. A scenario where the engine hasn't consumed one buffered event in 50ms would indicate a fundamentally broken event loop -- a much larger bug than what this test covers. Adding production-code synchronization hooks just for this edge case would be invasive and couple the test to implementation details.

**Review by josedonizetti — APPROVED** [review]

**Review by trvll — APPROVED** [review]
+1
